test/Test/MockApi.hs

```diff
@@ -75,21 +77,31 @@ data ExpectationRepetition
   | Always
   deriving (Eq, Ord, Show)
 
+data ExpectationCheckRequestValue
+  = Yes
+  | No
+  deriving (Eq, Ord, Show)
+
 -- | The result of a call can be either a value or a diagnostic failure
 newtype ApiResult a = ApiResult (Either ApiFail a)
 
 newtype ApiFail = ApiFail {unApiFail :: Text}
 
 -- | An expectation of an API call made up of the request and response.
 data ApiExpectation where
-  ApiExpectation :: ExpectationRepetition -> FossaApiClientF a -> ApiResult a -> ApiExpectation
+  ApiExpectation :: ExpectationRepetition -> ExpectationCheckRequestValue -> FossaApiClientF a -> ApiResult a -> ApiExpectation
```

Comment:
I might actually make this a different type here.  Maybe we have `ExactRequestExpectation` and `AnyRequestExpectation`?  I think the fact that we have to include a `FossaApiClientF a` is sort of accidental.

The other option would be to define some sort of "matcher".  Then instead of a `FossaApiClientF a` it would include a `RequestMatcher` which could have `Exact` and `Any` constructors.  Then we could even pass those into the first argument of `expectOnce` so you get something like 

```haskell
Exact (GetOrganization) `alwaysReturns` resp1
Any (GetProject ...) `returnsOnce` resp2
```

You could even do something cool with a typeclasses so you don't have to write `Exact`.

```haskell
class IsMatchableRequest f where
  toMatcher :: f a -> RequestMatcher a
  
instance IsMatchableRequest RequestMatcher where
  toMatcher = id

instance IsMachableRequest (FossaApiClientF a) where
  toMatcher = Exact

-- Not the real signature, just building the expectation for simplicity
returnsOnce :: IsMatchableRequest f => f a -> a -> ApiExpectation
returnsOnce matchable resp =
  ApiExpectation Once (toMatcher machable) resp
```